### PR TITLE
Fix a bug with SyntaxError stacktraces in Chrome

### PR DESCRIPTION
**Pull request by bengourley**
## Goal

`SyntaxError`s in Chrome have an `error.stack` property which is just `${err.name}: ${err.message}` and no stack frames, but because the `err.stack` value is truthy we attempted to parse it with `error-stack-parser`.

The result of this was that the filename property of the stackframes was not correct and caused issues with grouping. This change causes Report.getStacktrace(err) to generate a backtrace instead and then the file/line/col properties get filled in with the values from `window.onerror`.

### Added

A failing end-to-end test.

### Changed

The `hasStack(obj)` function which is used to determine whether we should parse the stacktrace now returns `false` if `obj.stack === ${err.name}: ${err.message}`. This causes the notifier to extract more information from the arguments which are passed to `window.onerror`, including filename, which means the stacktrace is correct in Chrome, matching that of other browsers.

## Tests

More detailed assertions about stackframes were added to the existing unhandled SyntaxError test case.

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: features/steps/browser_steps.rb [hunk 1]
```diff
@@ -31,6 +31,9 @@
   if err['columnNumber']
     step("the \"columnNumber\" of stack frame 0 equals #{err['columnNumber']}")
   end
+  if err['file']
+    step("the \"file\" of stack frame 0 ends with \"#{err['file']}\"")
```
Comment (bengourley) [inline features/steps/browser_steps.rb]: this is an "ends with" match so that the hard coded values don't need to contain the hostname/port/protocol
